pkg/identityserver/gateway_registry.go

```diff
@@ -57,6 +60,28 @@ func (is *IdentityServer) createGateway(ctx context.Context, req *ttnpb.CreateGa
 	if err := validateContactInfo(req.Gateway.ContactInfo); err != nil {
 		return nil, err
 	}
+	if len(req.FrequencyPlanIDs) == 0 && req.FrequencyPlanID != "" {
+		req.FrequencyPlanIDs = []string{req.FrequencyPlanID}
+	}
+
+	// Ensure that all Frequency Plan IDs are from the same band.
+	lenFPIDs := len(req.FrequencyPlanIDs)
+	if lenFPIDs > 1 {
+		for i := 0; i < (lenFPIDs - 1); i++ {
+			fp1, err := is.FrequencyPlans.GetByID(req.FrequencyPlanIDs[i])
+			if err != nil {
+				return nil, err
+			}
+			fp2, err := is.FrequencyPlans.GetByID(req.FrequencyPlanIDs[i+1])
+			if err != nil {
+				return nil, err
+			}
+			if fp1.BandID != fp2.BandID {
+				return nil, errFrequencyPlans
+			}
+		}
+	}
```

Comment:
I'm not sure the Identity Server should do this kind of validation. Doing that would prevent us from having (test) clusters that support experimental frequency plans.

If we do want to do this, I would write it as:

```go
if len(req.FrequencyPlanIDs) > 0 {
	fp0, err := is.FrequencyPlans.GetByID(req.FrequencyPlanIDs[0])
	if err != nil {
		return nil, err
	}
	for i := 1; i < len(req.FrequencyPlanIDs); i++ {
		fpn, err := is.FrequencyPlans.GetByID(req.FrequencyPlanIDs[i])
		if err != nil {
			return nil, err
		}
		if fpn.BandID != fp0.BandID {
			return nil, errFrequencyPlans
		}
	}
}
```

instead of fetching two frequency plans in every iteration of the loop

Reply:
@johanstokking ?

Reply:
I don't think we should be enforcing this in ER indeed; it doesn't have any authority in frequency plans.

Reply:
Ok? Then did I mis understand your comment here:https://github.com/TheThingsNetwork/lorawan-stack/pull/1826#discussion_r364665915?